components/script/dom/subtlecrypto.rs

```diff
@@ -860,6 +864,268 @@ impl SubtleCryptoMethods for SubtleCrypto {
+                    return;
+                }
+
+                let exported_key = match subtle.export_key_aes(format, &key) {
+                    Ok(k) => k,
+                    Err(e) => {
+                        promise.reject_error(e);
+                        return;
+                    },
+                };
+
+                let bytes = match exported_key {
+                    AesExportedKey::Raw(k) => k,
+                    AesExportedKey::Jwk(key) => {
+                        // The spec states to convert this to an ECMAscript object and stringify it, but since we know
+                        // that the output will be a string of JSON we can just construct it manually
+                        // TODO: Support more than just a subset of the JWK dict, or find a way to
+                        // stringify via SM internals
+                        let Some(k) = key.k else {
+                            promise.reject_error(Error::Syntax);
+                            return;
+                        };
+                        let Some(alg) = key.alg else {
+                            promise.reject_error(Error::Syntax);
+                            return;
+                        };
+                        let Some(ext) = key.ext else {
+                            promise.reject_error(Error::Syntax);
+                            return;
+                        };
+                        format!("{{
+                            \"kty\": \"oct\",
+                            \"k\": \"{}\",
+                            \"alg\": \"{}\",
+                            \"ext\": {}
+                        }}", k, alg, ext)
+                        .into_bytes()
+                    },
+                };
+
+                let cx = GlobalScope::get_cx();
+                rooted!(in(*cx) let mut array_buffer_ptr = ptr::null_mut::<JSObject>());
+
+                let result = match wrapping_alg_name.as_str() {
+                    ALG_AES_KW => {
+                        subtle.wrap_key_aes_kw(&wrapping_key, &bytes, cx, array_buffer_ptr.handle_mut())
+                    }
+                    ALG_AES_CBC => {
+                        let KeyWrapAlgorithm::AesCbc(params) = normalized_algorithm else {
+                            promise.reject_error(Error::Data);
+                            return;
+                        };
+                        subtle.encrypt_aes_cbc(&params, &wrapping_key, &bytes, cx, array_buffer_ptr.handle_mut())
+                    },
+                    ALG_AES_CTR => {
+                        let KeyWrapAlgorithm::AesCtr(params) = normalized_algorithm else {
+                            promise.reject_error(Error::Data);
+                            return;
+                        };
+                        subtle.encrypt_decrypt_aes_ctr(
+                            &params, &wrapping_key, &bytes, cx, array_buffer_ptr.handle_mut()
+                        )
```

Comment:
I believe the `let-else` branches are impossible because of the assertion above that the key name matches the provided algorithm (Step 8 in the spec). Can you match on `normalized_algorithm` instead of `wrapping_alg_name.as_str()`?

Reply:
Ah good catch, I'll adjust that